File: lib/services/api_service.dart
import 'dart:convert';
import 'package:http/http.dart' as http;

class ApiService {
  // Android Emulator
  static const String baseUrl = 'http://10.132.248.185:3000';

  // Get user by ID
  static Future<Map<String, dynamic>> getUser(int userId) async {
    final response = await http.get(
      Uri.parse('$baseUrl/api/users'),
    );

    if (response.statusCode == 200) {
      return jsonDecode(response.body);
    } else {
      throw Exception('Failed to load users');
    }
  }

  // Get user by Wallet ID
  static Future<Map<String, dynamic>> getUserByWalletId(
    String walletId,
  ) async {
    final response = await http.get(
      Uri.parse('$baseUrl/api/users/wallet/$walletId'),
    );

    if (response.statusCode == 200) {
      return jsonDecode(response.body);
    } else {
      throw Exception('User not found');
    }
  }

  // Get transactions for a user
  static Future<Map<String, dynamic>> getUserTransactions(
    int userId,
  ) async {
    final response = await http.get(
      Uri.parse('$baseUrl/api/transactions/user/$userId'),
    );

    if (response.statusCode == 200) {
      return jsonDecode(response.body);
    } else {
      throw Exception('Failed to load transactions');
    }
  }

  // Wallet to Wallet payment
  static Future<Map<String, dynamic>> walletTransfer({
  required String senderWalletId,
  required String receiverWalletId,
  required double amount,
  String? description,
}) async {

  
  final response = await http.post(
    Uri.parse(
      '$baseUrl/api/transactions/wallet-transfer',
    ),
    headers: {
      'Content-Type': 'application/json',
    },
    body: jsonEncode({
      'senderWalletId': senderWalletId,
      'receiverWalletId': receiverWalletId,
      'amount': amount,
      'description': description,
    }),
  );

  final data = jsonDecode(response.body);

  if (response.statusCode == 201) {
    return data;
  } else {
    throw Exception(
      data['message'] ?? 'Wallet transfer failed',
    );
  }
}}

File: _rn_legacy_/services/apiService.js
const BASE_URL = 'http://172.16.5.25:3000';

export const ApiService = {
  // Get user by Wallet ID
  getUserByWalletId: async (walletId) => {
    try {
      const response = await fetch(`${BASE_URL}/api/users/wallet/${walletId}`);
      if (response.ok) {
        return await response.json();
      }
      throw new Error('User not found');
    } catch (error) {
      console.warn('ApiService.getUserByWalletId error:', error.message);
      return null;
    }
  },

  // Get user transactions
  getUserTransactions: async (userId) => {
    try {
      const response = await fetch(`${BASE_URL}/api/transactions/user/${userId}`);
      if (response.ok) {
        return await response.json();
      }
      throw new Error('Failed to load transactions');
    } catch (error) {
      console.warn('ApiService.getUserTransactions error:', error.message);
      return null;
    }
  },

  // Wallet to Wallet Transfer
  walletTransfer: async ({ senderWalletId, receiverWalletId, amount, description }) => {
    try {
      const response = await fetch(`${BASE_URL}/api/transactions/wallet-transfer`, {
        method: 'POST',
        headers: {
          'Content-Type': 'application/json',
        },
        body: JSON.stringify({
          senderWalletId,
          receiverWalletId,
          amount,
          description,
        }),
      });

      const data = await response.json();
      if (response.status === 201) {
        return data;
      } else {
        throw new Error(data.message || 'Wallet transfer failed');
      }
    } catch (error) {
      console.warn('ApiService.walletTransfer error:', error.message);
      throw error;
    }
  },
};
